src/main/java/com/mraof/minestuck/command/DebugLandsCommand.java

```diff
@@ -32,7 +43,71 @@ public static void register(CommandDispatcher<CommandSourceStack> dispatcher)
 	private static int createDebugLands(CommandSourceStack source, List<LandTypePair> landTypes) throws CommandSyntaxException
 	{
 		ServerPlayer player = source.getPlayerOrException();
-		CommandActionHandler.createDebugLandsChain(player, landTypes, source);
+		createDebugLandsChain(player, landTypes);
 		return 1;
 	}
+	
+	public static void createDebugLandsChain(ServerPlayer player, List<LandTypePair> landTypes) throws CommandSyntaxException
+	{
+		var connections = SburbConnections.get(player.server);
+		PlayerIdentifier playerId = IdentifierHandler.encode(player);
+		
+		if(!SburbPlayerData.get(playerId, player.server).hasEntered())
+			throw MUST_ENTER_EXCEPTION.create();
+		
+		connections.unlinkClientPlayer(playerId);
+		connections.unlinkServerPlayer(playerId);
+		
+		PlayerIdentifier lastPlayer = playerId;
+		int i = 0;
+		for(; i < landTypes.size(); i++)
+		{
+			LandTypePair land = landTypes.get(i);
+			if(land == null)
+				break;
+			
+			PlayerIdentifier fakePlayer = IdentifierHandler.createNewFakeIdentifier();
+			
+			connections.newServerForClient(lastPlayer, fakePlayer);
+			connections.setPrimaryConnectionForEntry(fakePlayer);
+			
+			createAndSetLand(land, fakePlayer, player.server);
+			
+			lastPlayer = fakePlayer;
+		}
+		
+		if(i == landTypes.size())
+			connections.newServerForClient(lastPlayer, playerId);
+		else
+		{
+			PlayerIdentifier lastIdentifier = playerId;
+			for(i = landTypes.size() - 1; i >= 0; i--)
+			{
+				LandTypePair land = landTypes.get(i);
+				if(land == null)
+					break;
+				
+				PlayerIdentifier fakePlayer = IdentifierHandler.createNewFakeIdentifier();
+				
+				connections.setPrimaryConnection(fakePlayer, lastIdentifier);
+				createAndSetLand(land, fakePlayer, player.server);
+				
+				lastIdentifier = fakePlayer;
+			}
+		}
```

Comment:
This is a rather subjective comment but this block seems awkwardly written to me. It is not hard to read, just seems like their may be a more straightforward solution

Reply:
I agree. While I do think it's overall better than it was before, the looping/iteration here can still be made better.